### PR TITLE
BUGFIX: RAIL-4519 Attribute metadata fetching for deprecated attributes

### PR DESCRIPTION
**Pull request by plavemar**
- refactoring of useParentsConfiguration hook
- add query to fetch attributes for given display forms
- refactor handling of changes in parent filters configuration

JIRA: RAIL-4519

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9933/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11857/ : SUCCESS in 15m 41s

File: libs/sdk-ui-dashboard/src/model/queryServices/queryAttributeByDisplayForm.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    const { backend, workspace } = ctx;
+    const attribute = catalogAttributes.find((catalogAttribute) =>
+        catalogAttribute.displayForms.some((df) => areObjRefsEqual(df, displayForm)),
+    );
+
+    if (attribute) {
+        return attribute.attribute;
+    }
+
+    return await backend.workspace(workspace).attributes().getAttributeByDisplayForm(displayForm);
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/queryServices/queryAttributeByDisplayForm.ts:49]: Nitpick: Unless it is in a try catch block, you do not need `return await x`, just `return x` is sufficient.
The next function has also this redundancy.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3027

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3027

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2843/ : SUCCESS in 1m 16s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3148/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1989/ : SUCCESS in 20m 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11869/ : SUCCESS in 15m 41s

**Pull request merged**